.github/workflows/dotnet.yml

```diff
@@ -0,0 +1,44 @@
+name: .NET
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+jobs:
+  build:
+    runs-on: windows-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install .NET 6.0
+        uses: actions/setup-dotnet@v2
+        with:
+          dotnet-version: "6.0.x"
+      - name: Install .NET 5.0
+        uses: actions/setup-dotnet@v2
+        with:
+          dotnet-version: "5.0.x"
+      - name: Install .NET Core 3.1
+        uses: actions/setup-dotnet@v2
+        with:
+          dotnet-version: "3.1.x"
+      - name: Install .NET Core 2.1
+        uses: actions/setup-dotnet@v2
+        with:
+          dotnet-version: "2.1.x"
```

Comment:
If we did want to leave them all in, it might be preferable to do them in one action:
      - name: Install .NET Core 3.1
        uses: actions/setup-dotnet@v2
        with:
            dotnet-version: | 
              3.1.x
              5.0.x
etc.